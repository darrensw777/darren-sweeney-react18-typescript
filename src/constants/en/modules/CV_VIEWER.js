export const CV_VIEWER = {
    DOWNLOAD: "Click here to download a PDF of my CV",
    CONTACT: "Contact me",
    PAGE: "Page",
    OF: "of",
    NEXT: "Next",
    PREVIOUS: "Previous",
};
