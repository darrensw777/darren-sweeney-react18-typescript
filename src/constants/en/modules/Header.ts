export const HeaderCopy = {
    headerLinks: [
        {
            linkTo: '/',
            label: 'Home',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: true,
        },
        {
            linkTo: '/about',
            label: 'About',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/work',
            label: 'Work',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/music',
            label: 'Music',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/videos',
            label: 'Videos',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/gallery',
            label: 'Gallery',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/contact',
            label: 'Contact',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: true,
        },
    ],
};

export default HeaderCopy;
