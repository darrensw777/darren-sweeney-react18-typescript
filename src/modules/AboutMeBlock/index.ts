/* istanbul ignore file */
export { default } from './AboutMeBlock';
