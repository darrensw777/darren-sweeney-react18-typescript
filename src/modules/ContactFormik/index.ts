/* istanbul ignore file */
export { default } from './ContactFormik';
