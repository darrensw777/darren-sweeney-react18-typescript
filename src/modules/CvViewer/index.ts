/* istanbul ignore file */
export { default } from './CvViewer';
