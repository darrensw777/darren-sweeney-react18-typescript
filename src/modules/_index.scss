@import './Header/Header.scss';
@import './Footer/Footer.scss';
@import './PageHeader/page-header.scss';
@import './AboutMeBlock/about-me-block.scss';
@import './CvViewer/cv-viewer.scss';
@import './SoundCloudPlayer/soundcloud-player.scss';
@import './ImagesGallery/images-gallery.scss';
@import './ContactFormik/contact-formik.scss';
@import './InterestLists/interest-lists.scss';