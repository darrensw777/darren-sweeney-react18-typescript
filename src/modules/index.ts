/* istanbul ignore file */
/*
 * Exporting all the components from the components folder,
 * components can be imported as the example below:
 * import { Logo, Avatar } from 'components';
 */

export { default as Header } from './Header';
export { default as Footer } from './Footer';
export { default as PageHeader } from './PageHeader';
export { default as AboutMeBlock } from './AboutMeBlock';
export { default as CvViewer } from './CvViewer';
export { default as SoundCloudPlayer } from './SoundCloudPlayer';
export { default as ImagesGallery } from './ImagesGallery';
export { default as ContactFormik } from './ContactFormik';
export { default as InterestLists } from './InterestLists';
export { default as HomePageBlockContainer } from './HomePageBlockContainer';
