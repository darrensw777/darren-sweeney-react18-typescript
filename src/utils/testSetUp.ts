import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';

const testSetUp = ({ Component }: { Component: ReactElement }) => {
    const { container, getByText, getByRole, asFragment, getAllByRole, getByTestId, getByTitle } = render(Component, {
        wrapper: MemoryRouter,
    });
    return {
        container,
        getByText,
        getByRole,
        asFragment,
        getAllByRole,
        getByTestId,
        getByTitle,
    };
};

export default testSetUp;
