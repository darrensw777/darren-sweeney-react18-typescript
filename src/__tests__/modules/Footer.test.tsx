import { Footer } from 'modules';
import testSetUp from 'utils/testSetUp';

describe('Footer module', () => {
    const currentYear = new Date().getFullYear();

    let getByText: any, asFragment: any, container: any;
    beforeEach(() => {
        ({ asFragment, getByText, container } = testSetUp({ Component: <Footer /> }));
    });

    test('matches snapshot', () => {
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the footer', () => {
        const footer = container.getElementsByClassName('footer');
        expect(footer.length).toBe(1);
    });

    test('renders the copyright year', () => {
        expect(getByText(`Â©${currentYear} Darren Sweeney`)).toBeInTheDocument();
    });
});
