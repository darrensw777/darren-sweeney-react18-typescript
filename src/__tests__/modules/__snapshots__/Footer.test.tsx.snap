// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer module matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="moduleStyles__FooterWrapper-sc-187oehp-3 hAXTKZ"
  >
    <div
      class="footer"
      role="contentinfo"
    >
      <div
        class="inner-content"
      >
        <div
          class="copyright"
        >
          Â©2022 Darren Sweeney
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
